src/socket.rs

```diff
@@ -166,3 +168,31 @@ mod linux {
         <u32>::from_be(i)
     }
 }
+
+const EINVAL: i32 = 22;
+pub async fn relay(
+    downstream: &mut tokio::net::TcpStream,
+    upstream: &mut tokio::net::TcpStream,
+) -> Result<(), Error> {
+    #[cfg(all(target_os = "linux"))]
+    {
+        match realm_io::bidi_zero_copy(downstream, upstream).await {
+            Ok(()) => Ok(()),
+            Err(ref e) if e.raw_os_error().map_or(false, |ec| ec == EINVAL) => {
```

Comment:
This is referred from https://github.com/zhboner/realm/blob/master/realm_core/src/tcp/plain.rs

Reply:
any idea when it can occur on Linux?

Reply:
```
       EINVAL The target filesystem doesn't support splicing.

       EINVAL The target file is opened in append mode.

       EINVAL Neither of the file descriptors refers to a pipe.

       EINVAL An offset was given for nonseekable device (e.g., a pipe).

       EINVAL fd_in and fd_out refer to the same pipe.

```

Reply:
this is from `man splice`

Reply:
hmm.. are any of those even possible for our usage?

Reply:
I am not sure, just in case